### PR TITLE
Faster FileAccess::get_line()

### PR DESCRIPTION
**Pull request by ghost**
Startup with current master:

profile-master

Startup with this PR:

profile-pr

The idea of `CharBuffer` is similar to `StringBuffer` for bytes: it keeps a `stack_buffer` that avoids allocations for short buffers; `PoolVector` and allocating larger chunks avoids the `push_back` hit the current implementation has.

I also investigated a dedicated `LineReader` that internally reads chunks via `get_buffer`, but it needs its callers to change and it's not faster than the current change in the existing API.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
Can I just say that I like your efforts on optimizing Godot? :)

File: core/os/file_access.cpp [hunk 1]
```diff
@@ -273,9 +273,72 @@ String FileAccess::get_token() const {
...
+		}
+
+		buffer = vector_write.ptr();
+		capacity = vector.size();
+
+		return true;
+	}
+
+public:
+	_FORCE_INLINE_ CharBuffer()
```
Comment (akien-mga) [inline core/os/file_access.cpp]: Small style issue here (well your style is fine, but not consistent with that enforced by `clang-format` 5.0.0): https://travis-ci.org/godotengine/godot/jobs/316057663

**Comment by poke1024** [comment]
Fixed style; simplified call to `PoolVector::resize`

**Comment by reduz** [comment]
Not against it, but i would user regular Vector<> for this, since I think PoolVector is designed more for much larger buffers

**Comment by poke1024** [comment]
Changed to `Vector<>`.

**Pull request merged**